socket-server/internal/ws/server.go

```
@@ -33,38 +38,139 @@ func RunServer(w http.ResponseWriter, r *http.Request) {
+		log.Printf("UserId %s violate supported language", userId)
+		return
+	}
+
 	conn, err := upgrader.Upgrade(w, r, nil)
 	if err != nil {
 		log.Println("Upgrade error:", err)
 		// no need to write an error response here, as the upgrade has already failed
 		return
 	}
+	
+	assemblyConn, res, err := ConnectToAssemblyAI(config.EnvVars.AssemblyApiKey)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)